client/python/datajunction/client.py

```diff
@@ -1,1135 +1,15 @@
-    ):
-        """
-        Retrieves data for this node, given the provided dimensions and filters.
-        """
-        return self.dj_client.data(  # pragma: no cover
-            [self.name],
-            dimensions,
-            filters,
-            engine_name,
-            engine_version,
-        )
-
-    def deactivate(self) -> str:
-        """
-        Deactivates the node
-        """
-        response = self.dj_client.deactivate_node(self)
-        if not response.ok:  # pragma: no cover
-            raise DJClientException(
-                f"Error deactivating node `{self.name}`: {response.text}",
-            )
-        return f"Successfully deactivated `{self.name}`"
-
-    def activate(self) -> str:
-        """
-        Activates the node
-        """
-        response = self.dj_client.activate_node(self)
-        if not response.ok:  # pragma: no cover
-            raise DJClientException(
-                f"Error activating node `{self.name}`: {response.text}",
-            )
-        return f"Successfully activated `{self.name}`"
-
-    def revisions(self):
-        """
-        List all revisions of this node
-        """
-        return self.dj_client.get_node_revisions(self.name)
-
-    def add_availability(self, availability: models.AvailabilityState):
-        """
-        Adds an availability state to the node
-        """
-        return self.dj_client.add_availability_state(self.name, availability)
-
-    def set_column_attributes(self, attributes: List[models.ColumnAttribute]):
-        """
-        Sets attributes for columns on the node
-        """
-        return self.dj_client.set_column_attributes(self.name, attributes)
-
-
-class Source(Node):
+class DJClient(_DJClient):
     """
-    DJ source node
+    Client class to consume basic DJ services: metrics and dimensions.
     """
 
-    type: str = "source"
-    catalog: str
-    schema_: str
-    table: str
-    columns: Optional[List[models.Column]]
-
-    @validator("catalog", pre=True)
-    def parse_cls(  # pylint: disable=no-self-argument
-        cls,
-        value: Union[str, Dict[str, Any]],
-    ) -> str:
-        """
-        When `catalog` is a dictionary, parse out the catalog's
-        name, otherwise just return the string.
-        """
-        if isinstance(value, str):
-            return value
-        return value["name"]
-
-    def update(self) -> Dict:
-        """
-        Update the node for fields that have changed
-        """
-        update_node = models.UpdateNode(
-            display_name=self.display_name,
-            description=self.description,
-            mode=self.mode,
-            primary_key=self.primary_key,
-            catalog=self.catalog,
-            schema_=self.schema_,
-            table=self.table,
-            columns=self.columns,
-        )
-        return self.dj_client.update_node(self.name, update_node)
-
 
-class NodeWithQuery(Node):
+class DJBuilderClient(DJClient):
```

Comment:
Other naming options:
- `DJDeveloperClient`
- `DJModelerClient`
- `DJWriteClient`

Reply:
Looking at that, I wonder if we could drop the DJ prefix since the import makes it clear where it's coming from. That might make it a little cleaner, i.e. `from datajunction import ModelerClient`

Reply:
@shangyian :
- DJDeveloperClient  --> I considered that but it seems too far on the SWE side
- DJModelerClient --> yes, that's a good alternative to Builder
- DJWriteClient --> not sure if you meant Writer, but either way it feels "not enough"

@samredai, yes I considered doing that but, I think the prefix does not hurt, and it make the imported module "speak for itself". I think that a DJClient module/class or DJModelerClient maintain the context better.

Actually, what do you all think about dropping the Client suffix:
- **DJClient** for the consumption access
- **DJBuilder** or **DJModeler** for the contribution side

Alternatively: **DJReader** and **DJWriter** - hard to go wrong on this one. ;)

Reply:
Hmm, yeah I like `DJReader` and `DJWriter`!

Reply:
+1 to `DJReader` and `DJWriter`